src/components/calculator/Calculator.js

```diff
@@ -1,101 +1,83 @@
+  const updateCurrentOperation = ({ total, next, operation }) => {
+    const op = currentOp;
     if (total || total === null) {
-      currentOp.total = total;
+      op.total = total;
     }
     if (next || next === null) {
-      currentOp.next = next;
+      op.next = next;
     }
     if (operation || operation === null) {
-      currentOp.operation = operation;
+      op.operation = operation;
+    }
+    return op;
+  };
+
+  const cleanState = () => {
+    setOperation({ next: 0, total: null, operation: null });
+    setDelegate(0);
+  };
+
+  const setCalculatorResult = (value) => {
+    const op = currentOp;
+    try {
+      const results = calculate(op, value);
+      const { next, total, operation } = updateCurrentOperation(results);
+      setOperation({ next, total, operation });
+    } catch (error) {
+      setOperation({ next: 'error', total: null, operation: null });
+      setDelegate(1);
     }
-    return currentOp;
-  }
+  };
+
+  const doFunction = (param) => {
+    const delegateFunctions = [setCalculatorResult, cleanState];
+
+    delegateFunctions[delegate](param);
+  };
 
-  render() {
-    const {
-      numbers, decimalDot, special, operators, currentOp,
-    } = this.state;
-    const numBtns = numbers.map((num) => (
+  const clickHandler = (e) => {
+    doFunction(e.target.value);
+  };
+
+  const createButtons = (buttonCollection) => {
+    const btns = buttonCollection.map((num) => (
       <CalculatorButton
         key={num}
         number={num.toString()}
-        clickHandler={this.clickHandler}
-      />
-    ));
-    const specialBtns = special.map((btn) => (
-      <CalculatorButton
-        key={btn}
-        number={btn.toString()}
-        clickHandler={this.clickHandler}
+        clickHandler={clickHandler}
       />
     ));
-    const operatorBtns = operators.map((btn) => (
-      <CalculatorButton
-        key={btn}
-        number={btn.toString()}
-        clickHandler={this.clickHandler}
-      />
-    ));
-    return (
-      <div className="main-container">
-        <Display number={currentOp} />
-        <ul className="special-container">
-          {specialBtns}
-        </ul>
-        <ul className="numbers-container">
-          {numBtns}
-          <CalculatorButton
-            className="dot-btn"
-            type="button"
-            clickHandler={this.clickHandler}
-            number={decimalDot}
-          />
-        </ul>
-        <ul className="operator-container">
-          {operatorBtns}
-        </ul>
-      </div>
-    );
-  }
+    return btns;
+  };
+
+  return (
+    <div className="main-container">
+      <Display value={currentOp} />
+      <ul className="special-container">
+        {createButtons(specialKeys)}
+      </ul>
+      <ul className="numbers-container">
+        {createButtons(numberKeys)}
+        {createButtons(decimalKey)}
+      </ul>
+      <ul className="operator-container">
+        {createButtons(operatorKeys)}
+      </ul>
+    </div>
+  );
 }
```

Comment:
- [ ] While you have done a great job with the project and change class to function but I would recommend you to use ES6 sntax. Please work on this change.

Reply:
Done